google/cloud/aiplatform/models.py

```diff
@@ -215,10 +228,483 @@ def upload(
+                If a DeployedModel's ID is not listed in this map, then it receives
+                no traffic. Key for model being deployed is "0". Should not be
+                provided if traffic_percentage is provided.
+            machine_type (str):
+                Optional. The type of machine. Not specifying machine type will
+                result in model to be deployed with automatic resources.
+            min_replica_count (int):
+                Optional. The minimum number of machine replicas this deployed
+                model will be always deployed on. If traffic against it increases,
+                it may dynamically be deployed onto more replicas, and as traffic
+                decreases, some of these extra replicas may be freed.
+            max_replica_count (int):
+                Optional. The maximum number of replicas this deployed model may
+                be deployed on when the traffic against it increases. If requested
+                value is too large, the deployment will error, but if deployment
+                succeeds then the ability to scale the model to that many replicas
+                is guaranteed (barring service outages). If traffic against the
+                deployed model increases beyond what its replicas at maximum may
+                handle, a portion of the traffic will be dropped. If this value
+                is not provided, the smaller value of min_replica_count or 1 will
+                be used.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+        Returns:
+            lro (lro.LRO):
+                LRO of model deployment.
+        """
+        max_replica_count = min(1, min_replica_count)
+        if machine_type:
+            machine_spec = machine_resources.MachineSpec(machine_type=machine_type)
+            dedicated_resources = machine_resources.DedicatedResources(
+                machine_spec=machine_spec,
+                min_replica_count=min_replica_count,
+                max_replica_count=max_replica_count,
+            )
+            deployed_model = gca_endpoint.DeployedModel(
+                dedicated_resources=dedicated_resources,
+                model=model.resource_name,
+                display_name=deployed_model_display_name,
+            )
+        else:
+            automatic_resources = machine_resources.AutomaticResources(
+                min_replica_count=min_replica_count,
+                max_replica_count=max_replica_count,
+            )
+            deployed_model = gca_endpoint.DeployedModel(
+                automatic_resources=automatic_resources,
+                model=model.resource_name,
+                display_name=deployed_model_display_name,
+            )
+
+        if traffic_split is None:
+            if bool(self._gca_resource.traffic_split) is False:
```

Comment:
Will self._gca_resource.traffic_split not automatically evaluate to False without the need of type casting?

Also this is equivalent to `if not bool(self._gca_resource.traffic_split)`